virttest/utils_disk.py

```diff
@@ -182,6 +188,583 @@ def clean_old_image(image):
+    """
+    create file system in windows or linux guest.
+
+    :param session: session object to guest.
+    :param partition_name: partition name that to be formatted.
+    :param fstype: file system type for the disk.
+    :param ostype: guest os type 'windows' or 'linux'.
+    """
+    if ostype == "windows":
+        create_filesystem_windows(session, partition_name, fstype)
+    else:
+        create_filesyetem_linux(session, partition_name, fstype)
+
+
+def set_drive_letter(session, did, partition_no=1):
+    """
+    set drive letter for partition in windows guest.
+
+    :param session: session object to guest.
+    :param did: disk index
+    :param partition_no: partition number
+    :return drive_letter: drive letter has been set
+    """
+    list_partition_cmd = ' echo list partition '
+    list_partition_cmd = wrap_windows_cmd(list_partition_cmd)
+    details = session.cmd_output(list_partition_cmd % did)
+    for line in details.splitlines():
+        if re.search("Reserved", line, re.I | re.M):
+            partition_no += int(line.split()[1])
+    assign_letter_cmd = ' echo select partition %s ; echo assign '
+    assign_letter_cmd = wrap_windows_cmd(assign_letter_cmd)
+    session.cmd(assign_letter_cmd % (did, partition_no))
+    detail_cmd = ' echo detail disk '
+    detail_cmd = wrap_windows_cmd(detail_cmd)
+    details = session.cmd_output(detail_cmd % did)
+    for line in details.splitlines():
+        pattern = "\s+Volume\s+\d+"
+        if re.search(pattern, line, re.I | re.M):
+            return line.split()[2]
+
+
+def drop_drive_letter(session, drive_letter):
+    """
+    remove drive letter for partition in windows guest.
+
+    :param session: session object to guest.
+    :param drive_letter: drive letter to be removed
+    """
+    disk = "disk_" + utils_numeric.generate_random_string(4)
+    remove_cmd = "echo select volume %s > " + disk
+    remove_cmd += " && echo remove >> " + disk
+    remove_cmd += " && echo exit >> " + disk
+    remove_cmd += " && diskpart /s " + disk
+    remove_cmd += " && del /f " + disk
+    session.cmd(remove_cmd % drive_letter)
+
+
+def format_windows_disk(session, did, size, n_partitions=1, fstype="ntfs",
+                        labeltype=PARTITION_TABLE_TYPE_MBR):
+    """
+    Create partition on disks in windows guest and format it.
+
+    :param session: session object to guest.
+    :param did: disk index which show in 'diskpart list disk'.
+    :param size: partition size.
+    :param n_partitions: the number of partitions on disk
+    :param fstype: filesystem type for the disk.
+    :param labeltype: label type for the disk.
+    :return a list: mount point list for all partitions.
+    """
+    mountpoint = []
+    create_partition_table_windows(session, did, labeltype)
+    create_partition_batch_windows(session, did, size, n_partitions, labeltype)
+    for i in range(n_partitions):
+        mountpoint.append(set_drive_letter(session, did, partition_no=i+1))
```

Comment:
If a partition failed to get a drive letter, we should stop to configure the next partition and report the error immediately.